### ISSUE TITLE
multi threading + print statement breaks tqdm

### ISSUE THREAD

**Issue by SRK7Kyros** [tqdm/tqdm#1412]
- [x] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [x] visual output bug
- [ ] I have visited the [source website], and in particular
  read the [known issues]
- [ ] I have searched through the [issue tracker] for duplicates
- [x] I have mentioned version numbers, operating system and
  environment, where applicable:
  ```python
  import tqdm, sys
  print(tqdm.__version__, sys.version, sys.platform)
  ```
tqdm: 4.64.0
sys: 3.9.12
platform: (main, Apr  4 2022, 05:22:27) [MSC v.1916 64 bit (AMD64)] win32

[source website]: https://github.com/tqdm/tqdm/
[known issues]: https://github.com/tqdm/tqdm/#faq-and-known-issues
[issue tracker]: https://github.com/tqdm/tqdm/issues?q=

Ok so basically I'm using tqdm to show progress when downloading a file.
My program downloads various different files concurrently, using the threading module.
So basically I create a Thread object for every file I want to download, add it to the queue, and then every task gets executed.
The function being run downloads the file, then at the end prints a message to the console.

Now
I think that while more than 1 tqdms are active in a console if a print statement occurs by one of those, it sort of breaks all the layout of the console, resulting in things like this: https://cdn.discordapp.com/attachments/1057766895970947082/1057778949041700974/image.png

How do I solve it?

**Comment by Lenormju** [comment]
You can have several progress bars in parallel : see [this answer on StackOverflow](https://stackoverflow.com/a/46398645/11384184) (you need a modern terminal and `colorama`). Use the `position` parameter for [`tqdm`](https://tqdm.github.io/docs/tqdm/#__init__), one for each file. You can set the `desc` too.

```python
import time
import threading
import tqdm

def fake_file_downloader(filename: str, pbar: tqdm.tqdm) -> None:
    random = hash(filename) % 100
    for _ in range(random):  # simulate downloading
        pbar.update(1)
        time.sleep(0.1)
    pbar.set_description(f"Finished downloading {filename!r}")
    pbar.close()

files = [
    "a.txt",
    "b.doc",
    "c.mp4",
    "d.css",
]
thread_by_file = {}
pbar_by_file = {}

for i, filename in enumerate(files):
    pbar = tqdm.tqdm(desc=f"Downloading {filename!r} ...", leave=False)
    pbar_by_file[filename] = pbar
    thread = threading.Thread(target=fake_file_downloader, args=(filename, pbar))
    thread_by_file[filename] = thread

for filename in files:
    thread_by_file[filename].start()
for filename in files:
    thread_by_file[filename].join()

print("all done")
```
```python
Downloading 'a.txt' ...: 28it [00:02,  9.99it/s]
Downloading 'b.doc' ...: 28it [00:02, 10.00it/s]
Downloading 'c.mp4' ...: 28it [00:02,  9.99it/s]
Downloading 'd.css' ...: 28it [00:02,  9.99it/s]
```
But because `leave=False` then they disappear, you can't positively see they are finished downloading.

If you set `leave=True` then the stdout gets messed up :  
```
Finished downloading 'b.doc': : 16it [00:01,  9.96it/s]
Finished downloading 'a.txt': : 31it [00:03,  9.97it/s]
Finished downloading 'c.mp4': : 43it [00:04,  9.96it/s]
Finished downloading 'd.css': : 51it [00:05,  9.96it/s]
all done downloading 'c.mp4': : 43it [00:04,  9.95it/s]
Downloading 'd.css' ...: 43it [00:04,  9.95it/s]
Process finished with exit code 01it [00:05,  9.95it/s]
```
When one file finishes, it gets moved to the top, the others are pushed down, but there seems to be an error here.

**Comment by SRK7Kyros** [comment]
Thank you @Lenormju for your reply :D
I’m going to tinker with your idea as soon as I get back home from vacation, but it looks promising. Actually i don’t really need the bars to stay there, now that I think about it. I can just print they’re done in the function im passing to the Thread. 
Ty :D
